### DISCUSSION TITLE
Cannot read properties of undefined (reading 'overrideDisplayStyle')

### DISCUSSION THREAD

**Discussion by VedankNaik** [iTwin/itwinjs-core#2802] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/47364632/143994891-d5d27795-6c7b-4dec-bd28-d452f137abb4.png)

Could not pinpoint in what cases this appears. Sometimes it appears on starting the project, while interacting with UI elements. What is going wrong?

**Comment by pmconne** [comment, marked as answer]
`IModelApp.viewManager.selectedView` is undefined. Most likely your code is running before `viewPort` is added to the ViewManager, though I can't tell for certain from the snippet you provided.
Change it to `viewPort.overrideDisplayStyle({backgroundColor: 0xEBEBEB});`.

Incidentally, making the map fully transparent is a very inefficient way of hiding the map - it's going to cause the app to continue to request map tiles and draw them invisibly.
To turn off the map:
```
const viewflags = viewPort.viewFlags.clone();
viewflags.backgroundMap = false;
viewPort.viewFlags = viewflags;
```

**Reply by VedankNaik** [reply]
Thanks, will keep watch if error occurs after making above changes. Thanks for map hiding code too.
